test_data/asm/generic_permutation.asm

```diff
@@ -0,0 +1,45 @@
+let N: int = 8;
+
+use std::array;
+use std::check;
+use std::convert::fe_wrapping;
+
+// TODO I think we do not have away to dynamically create a fixed column...
+
+let permutation = constr |first, left, right| {
+    std::check::assert_eq(array::len(left), array::len(right));
+
+    let alpha = std::prover::challenge(0, 1);
+    let beta = std::prover::challenge(0, 2);
+
+    let left_compressed = compress_selected_exprs(alpha, beta, 1, left);
+    let right_compressed = compress_selected_exprs(alpha, beta, 1, right);
+
+    let z = std::prover::new_witness_in_stage("permutation_accumulator", 1);
+
+    first * (z - 1) = 0;
+
+    // Update rule:
+    // z' = z * (beta - a) / (beta - b)
+    (beta - left_compressed) * z' = z * (beta - right_compressed);
+    // we need to return something...
+    []
+};
+
+let compress_selected_exprs: expr, expr, expr, expr[] -> expr = |alpha, beta, sel, exprs|
+    sel * (compress_columns(alpha, exprs) + beta - 1) + 1;
```

Comment:
```suggestion
    sel * (beta - compress_columns(alpha, exprs) - 1) + 1;
```
I think both is sound, but this makes it clear we're evaluating $$\prod_{i = 1}^n (X - a_i)$$ at a random point.